### PR TITLE
fix(ThemeProvider): inner provider should overwrite specified props and only them

### PR DESCRIPTION
**Pull request by ValeraS**

### CODE REVIEW

File: test-utils/utils.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    direction: 'ltr',
+    theme: 'light',
+    themeValue: 'light',
+};
+
+function Providers({children}: {children: React.ReactElement}) {
+    return <ThemeContext.Provider value={themeContextValue}>{children}</ThemeContext.Provider>;
+}
+
+const createWrapper = memoize(
```
Comment (ogonkov) [inline test-utils/utils.tsx:20]: Memoization is really necessary?

Comment (ValeraS) [reply test-utils/utils.tsx]: Taking a closer look, memoization is unnecessary here.

**Comment by ogonkov** [comment]
We are just add wrapper with theme context to tests here?

File: styles/themes/common/_index.scss [hunk 2]
```diff
@@ -14,4 +14,23 @@
     --g-border-radius-xl: 10px;
 
     --g-focus-border-radius: 2px;
+
+    @at-root body#{&} {
```
Comment (amje) [inline styles/themes/common/_index.scss:18]: The root can be on any element, not only the body. Also these styles should go into styles.scss, not here

**Review by amje — CHANGES_REQUESTED** [review]

File: styles/themes/common/_index.scss (hunk 2 above)
Comment (ValeraS) [reply styles/themes/common/_index.scss:18]: These defaults only for the class on the body, if present, any other element always has `dir` attribute if it set to `ThemeProvider`.

**Review by amje — APPROVED** [review]

**Pull request merged**